iot_verifier/src/poc.rs

```diff
@@ -771,19 +777,43 @@ mod tests {
     fn test_verify_beacon_schedule() {
         let now = Utc::now();
         let id: &str = "test_id";
-        let last_beacon_a = Some(LastBeacon {
+        let last_beacon = Some(LastBeacon {
             id: id.as_bytes().to_vec(),
-            timestamp: now - Duration::seconds(60 * 12),
+            timestamp: now - *BEACON_INTERVAL,
         });
-        // last beacon was 12 mins in the past, expectation pass
-        assert_eq!(Ok(()), verify_beacon_schedule(&last_beacon_a, now));
-        //we dont have any last beacon data, expectation pass
-        assert_eq!(Ok(()), verify_beacon_schedule(&None, now));
-        // too soon after our last beacon, expectation fail
+        // last beacon was BEACON_INTERVAL in the past, expectation pass
+        assert_eq!(Ok(()), verify_beacon_schedule(&last_beacon, now));
```

Comment:
instead of `assert_eq` here, you might want to do `assert!(verify_beacon_schedule(&last_beacon, now).is_ok())`

Reply:
updated all similar here: f26a924836c10144187fe55e19c28a3a9f1827c0